lib/elastic_record/from_search_hit.rb

```diff
@@ -19,40 +19,33 @@ def from_search_hit(hit, mapping_properties = elastic_index.mapping[:properties]
 
     private
 
-      def value_from_search_hit_object(hit, mapping_properties)
-        mapping_properties.each do |(field, mapping)|
-          value = value_from_search_hit hit, field, mapping, mapping_properties
-
-          unless value.nil?
-            hit[field] = value
+      def value_from_search_hit_object(hit)
+        hit.each do |field, value|
+          next unless value
+
+          case value
+          when Hash
+            hit[field] = value_from_search_hit(value)
+          when Array # type: 'nested'
+            value.each do |element|
+              if element.is_a? Hash
+                value_from_search_hit_object(element)
+              end
+            end
           end
         end
 
         hit
       end
 
-      def value_from_search_hit(hit, field, mapping, mapping_properties)
-        value = hit[field]
-        return if value.nil?
-
-        value =
-          case mapping[:type]&.to_sym
-          when :object
-            object_mapping_properties = mapping_properties.dig(field, :properties)
-            value_from_search_hit_object(value, object_mapping_properties)
-          when :nested
-            object_mapping_properties = mapping_properties.dig(field, :properties)
-            value.map { |entry| value_from_search_hit_object(entry, object_mapping_properties) }
-          when :integer_range, :float_range, :long_range, :double_range
-            value_for_range(value)
-          when :date_range
-            value_for_date_range(value)
-          else
-            value
-          end
-
-        if value.present? || value == false
-          value
+      def value_from_search_hit(value)
+        case value['gte'] # the gte lower bound is never nil
```

Comment:
Why is this happening?

Reply:
`value['gte']` was the most reliable value used to determine the correct handler. The previous approach was to check for both `gte` and `lte` keys in the value hash, but it triggered an unnecessary array.

Reply:
We need to infer the type of the key `lte`
* String type - `date_range`
* Integer type - `integer_range`